mcp/src/databricks_mcp/mcp.py

```diff
@@ -0,0 +1,130 @@
+import asyncio
+import logging
+import re
+from typing import Any, List, Optional
+from urllib.parse import urlparse
+
+from databricks.sdk import WorkspaceClient
+from mcp.client.session import ClientSession
+from mcp.client.streamable_http import streamablehttp_client
+from mcp.types import CallToolResult, Tool
+from mlflow.models.resources import (
+    DatabricksFunction,
+    DatabricksGenieSpace,
+    DatabricksResource,
+    DatabricksVectorSearchIndex,
+)
+
+from databricks_mcp.oauth_provider import DatabricksOAuthClientProvider
+
+logger = logging.getLogger(__name__)
+
+# MCP URL types
+UC_FUNCTIONS_MCP = "uc_functions_mcp"
+VECTOR_SEARCH_MCP = "vector_search_mcp"
+GENIE_MCP = "genie_mcp"
+
+MCP_URL_PATTERNS = {
+    UC_FUNCTIONS_MCP: r"^/api/2\.0/mcp/functions/[^/]+/[^/]+$",
+    VECTOR_SEARCH_MCP: r"^/api/2\.0/mcp/vector-search/[^/]+/[^/]+$",
+    GENIE_MCP: r"^/api/2\.0/mcp/genie/[^/]+$",
+}
+
+
+class DatabricksMCPClient:
```

Comment:
Can we annotate this class as experimental using https://github.com/databricks/databricks-ai-bridge/blob/49bc442aebc04a8ad5d4c40d6d9aba8eba42e07a/src/databricks_ai_bridge/utils/annotations.py#L31?